tracer/src/Datadog.Trace/IAST/IastModule.cs

```diff
@@ -89,5 +106,35 @@ private static bool InvalidHashAlgorithm(string algorithm, Iast iast)
 
             return false;
         }
+
+        private static bool InvalidCipherAlgorithm(Type type, string algorithm, Iast iast)
+        {
+            if (ProviderValid(type.Name))
+            {
+                return false;
+            }
+
+            foreach (var weakCipherAlgorithm in iast.Settings.WeakCipherAlgorithmsArray)
+            {
+                if (string.Equals(algorithm, weakCipherAlgorithm, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool ProviderValid(string name)
+        {
+            // TripleDESCryptoServiceProvider internally creates a DES algorithm instance.
+            if (name == "TripleDESCryptoServiceProvider" ||
+                (string.Equals(FrameworkDescription.Instance.OSPlatform, "Linux", StringComparison.OrdinalIgnoreCase) && name.ToLower().EndsWith("provider")))
+            {
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
nit: we can simplify this a little 🙂 
- we set the OSPlatform, so don't need to ignore case
- we have constants for it. 
- We can avoid the string allocation in the name comparison
- Can remove the `if`
- I love switch expressions 😄 

Though, I'm a bit confused - this seems to say that `TripleDes` amd Des are _valid_ providers, i.e. they _aren't_ weak. But then you explicitly have `DES,TRIPLEDES` in the weak cipher defaults 🤔 Is that correct? Are they weak or not? 😄 

```suggestion
        private static bool ProviderValid(string name)
            => name switch
            {
                // TripleDESCryptoServiceProvider internally creates a DES algorithm instance.
                "TripleDESCryptoServiceProvider" => true,
                string.Equals(FrameworkDescription.Instance.OSPlatform, OSPlatformName.Linux, StringComparison.Ordinal) && name.EndsWith("provider", StringComparison.OrdinalIgnoreCase)) => true,
                _ => false
            };
```

Reply:
TripleDES and DES are weak algorithms. The problem is that TripleDESCryptoServiceProvider is a SymetricAlgorithm that internally creates a TripleDES instance, which is also a weak SymmetricAlgorithm. In order to avoid launching two spans for a single vulnerability, we skip the one that would be launched when instantiating the TripleDESCryptoServiceProvider class.

Reply:
Ahhhh, ok, that makes sense, thanks! 🙂

Reply:
I think the suggested changes still make sense though + might be worth adding that description in the comments in case someone has the same thoughts? 🙂

Reply:
Sure! I have added it.